app/services/fees/create_recurring_pay_in_advance_service.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Fees
+  class CreateRecurringPayInAdvanceService < BaseService
+    def initialize(subscriptions:, billing_at:)
+      @subscriptions = subscriptions
+      @billing_at = billing_at
+
+      super
+    end
+
+    def call
+      plan_ids = subscriptions.select(&:active?).map(&:plan_id).uniq
+
+      eligible_charges(plan_ids).find_each do |charge|
+        last_fee = charge.fees.order(created_at: :desc).first
+        event = Event.find_by(id: last_fee&.pay_in_advance_event_id)
+
+        if event
+          Fees::CreatePayInAdvanceJob.perform_later(charge:, event:, billing_at:)
+        end
+      end
```

Comment:
Could you return the `result` event if no property is attached to it? It's just to keep the common pattern ;)